packages/charts/src/hooks/useLabelElements.ts

```diff
@@ -6,24 +6,33 @@ import {
   YAxisTicks
 } from '../internal/CustomElements';
 import { ThemingParameters } from '@ui5/webcomponents-react-base/lib/ThemingParameters';
-import { renderAxisTicks } from '../util/Utils';
+import { getTextWidth, renderAxisTicks } from '../util/Utils';
 
-export const useDataLabel = (dataLabel, dataLabelCustomElement, dataLabelFormatter, stacked?, bar?) =>
+export const useDataLabel = (dataLabel, dataLabelCustomElement, dataLabelFormatter, stacked?, bar?, noSizeCheck?) =>
   useMemo(() => {
-    return dataLabel
+    return dataLabel || typeof dataLabel === 'undefined'
       ? dataLabelCustomElement
         ? (props) => DataLabel(props, dataLabelFormatter, dataLabelCustomElement)
         : {
             position: bar ? (stacked ? 'insideRight' : 'right') : stacked ? 'inside' : 'top',
-            content: (props) => dataLabelFormatter(props.value),
+            content: (props) => {
+              const formattedDataValue = dataLabelFormatter(props.value);
+              if (props.viewBox.width < getTextWidth(formattedDataValue) && !noSizeCheck) {
```

Comment:
I think it would be simpler to have it like this extract the no size check as first if statement:
```js
if (noSizeCheck) {
  return formattedDataValue;
}
```